openshift_tools/ircbot/trello/trello.py

```diff
@@ -128,6 +150,97 @@ def get(self):
                                                  members)
         return results
 
+    def report(self, email=None, move=False):
+        """Generate reports"""
+        payload = self.report_payload()
+        print(payload)
+        _env_email = os.environ.get("trello_report_email_addresses", None)
+        if _env_email:
+            email = _env_email
+        if self.args:
+            if self.args.email:
+                email = self.args.email
+            if self.args.move:
+                move = self.args.move
+        week_number = date.today().isocalendar()[1]
+        if email:
+            subj = "OpenShift SRE Report for Week #{} (ending {})".format(
+                week_number, date.today().strftime("%d-%m-%Y"))
+            _board_url = "https://trello.com/b/{}".format(
+                os.environ.get("trello_board_id", None))
+            payload = "For more information visit the SRE 24x7 board {}\n\n{}".format(
+                _board_url, payload)
+            self.email_report(email, subj, payload)
+            print("Report emailed to {}".format(email))
+        if move:
+            list_name = "Week #{}".format(week_number)
+            self.move_cards(to_list=list_name)
+            print("Cards moved to list '{}'".format(list_name))
+
+    def report_payload(self):
+        """Return report payload
+        :return: formatted report"""
+        data = ""
+        resolved_cards = self.get_list_cards(DEFAULT_RESOLVED_LIST)
+        data += "{}: {}\n".format(DEFAULT_LIST,
+                                  len(self.get_list_cards(DEFAULT_LIST)))
+        data += "{}: {}\n".format(DEFAULT_SNOWFLAKE_LIST,
+                                  len(self.get_list_cards(DEFAULT_SNOWFLAKE_LIST)))
+        data += "{}: {}\n".format(DEFAULT_RESOLVED_LIST,
+                                  len(resolved_cards))
+        data += "\n---\nResolved issues:\n---\n"
+        for card in resolved_cards:
+            data += "{} {}\n".format(card['shortUrl'], card['name'])
+        return data
+
+    def move_cards(self, from_list=None, to_list=None):
+        """Move cards from one list to another
+        :param from_list: name of list to move card from
+        :param to_list: name of list to move cards to
+        :return: None"""
+        params = {}
+        if not from_list:
```

Comment:
No check if `to_list` exists. What happens if it tries to `self.create_list(None)`?

Reply:
Ahh, if I had kept reading...it looks like it just creates a list called "None"?

Reply:
Update pushed to handle this case. Thanks.

Reply:
:+1: Awesome!